drainer/translator/translator.go

```diff
@@ -74,7 +96,18 @@ func insertRowToDatums(table *model.TableInfo, row []byte) (pk types.Datum, datu
 			//
 			// Will get -1 here, note: uint64(int64(-1)) = 18446744073709551615
 			// so we change it to uint64 if the column type is unsigned
-			datums[col.ID] = fixType(pk, col)
+			datums[col.ID] = fixType(pk[0], col)
+		case clusteredPk:
+			for idxColOrdinal, idxCol := range commonPKInfo.Columns {
+				if idxCol.Length != types.UnspecifiedLength {
+					// primary key's prefixed column already in row data.
+					continue
+				}
+				if tblColOrdinal == idxCol.Offset {
```

Comment:
can we directly access `table.Columns[idxCol.Offset]`, to eliminate the two-level for loop? seems we only need to check `clusteredPK` per table.